### PR TITLE
Enforce that all scripts use Python 3 rather than Python 2

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request makes a change to enforce that all the scripts contained in this repo use Python 3 vice Python 2.

## 💭 Motivation and context ##

The one remaining script that used Python 2 code appears to work with either Python 2 or Python 3, so it makes sense to insist on Python 3 now that Python 2 is EOL.

## 🧪 Testing ##

All automated tests pass.

@mcdonnnj, is this one line change something that can be manually done in production to verify that it functions as expected?

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

**Comment by dav3r** [comment]
> @mcdonnnj, is this one line change something that can be manually done in production to verify that it functions as expected?

We should be able to test that in either @mcdonnnj or my CyHy dev environments.

**Comment by jsf9k** [comment]
FYI, the main branch of this project [is now failing APB builds](https://github.com/cisagov/cyhy-feeds/actions/runs/2586197733).

**Comment by jsf9k** [comment]
@mcdonnnj, could you leave a comment here explaining why this PR cannot yet be merged?

**Comment by jsf9k** [comment]
@mcdonnnj, could you leave a comment here explaining why this PR cannot yet be merged?

**Comment by mcdonnnj** [comment]
@jsf9k Until we're deploying Debian Buster/Bullseye instances we have to consider Debian Stretch which only has Python 3.5. The lowest version of Python 3 this supports is 3.6 per https://github.com/cisagov/cyhy-feeds/blob/d02e28e4fce54532f5d28182efc1c44dfee5ad2c/setup.py#L83 This PR can _hopefully_ be merged this week as I finalize testing and prepare to deploy Debian Buster/Bullseye instances.

**Comment by jsf9k** [comment]
> @jsf9k Until we're deploying Debian Buster/Bullseye instances we have to consider Debian Stretch which only has Python 3.5. The lowest version of Python 3 this supports is 3.6 per
> 
> https://github.com/cisagov/cyhy-feeds/blob/d02e28e4fce54532f5d28182efc1c44dfee5ad2c/setup.py#L83
> 
> This PR can _hopefully_ be merged this week as I finalize testing and prepare to deploy Debian Buster/Bullseye instances.

Now that [Debian Stretch is dead as Dillinger](https://github.com/cisagov/skeleton-ansible-role/pull/124), does this change anything?

**Comment by mcdonnnj** [comment]
> > @jsf9k Until we're deploying Debian Buster/Bullseye instances we have to consider Debian Stretch which only has Python 3.5. The lowest version of Python 3 this supports is 3.6 per
> > https://github.com/cisagov/cyhy-feeds/blob/d02e28e4fce54532f5d28182efc1c44dfee5ad2c/setup.py#L83
> > 
> > This PR can _hopefully_ be merged this week as I finalize testing and prepare to deploy Debian Buster/Bullseye instances.
> 
> Now that [Debian Stretch is dead as Dillinger](https://github.com/cisagov/skeleton-ansible-role/pull/124), does this change anything?

No. If there is a critical problem with the Buster images I can always create a cursed Ansible role to pull from the Debian archive for package updates to fall back on Stretch. I would rather minimize any peripheral impact until I get new images built, deployed, and verified.

> **Warning**
> I realized belatedly that this PR was never included in my batch of testing. I have a [branch in cisagov/ansible-role-cyhy-feeds](https://github.com/cisagov/ansible-role-cyhy-feeds/tree/improvement/install_as_python3_package) and am troubleshooting installing this package and its dependencies under Python 3.

**Review by jasonodoom — APPROVED** [review]

**Comment by mcdonnnj** [comment]
@dav3r @jsf9k Would you mind reviewing this PR now that I've added a number of changes. I tested this branch (with my updates) in my development CyHy environment and was able to confirm that it worked without errors.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -107,21 +103,26 @@ jobs:
...
         python-version:
-          - "2.7"
-          - "3.6"
           - "3.7"
           - "3.8"
           - "3.9"
           - "3.10"
+          - "3.11"
+        include:
+          - os: ubuntu-20.04
```
Comment (jsf9k) [inline .github/workflows/build.yml:120]: Why is this necessary?

File: setup.py [hunk 2]
```diff
@@ -9,45 +9,58 @@
...
     description="Data feed components for Cyber Hygiene",
     long_description=readme(),
     long_description_content_type="text/markdown",
+    # Landing page for CISA's cybersecurity mission
+    url="https://www.cisa.gov/cybersecurity",
+    # Additional URLs for this project per
+    # https://packaging.python.org/guides/distributing-packages-using-setuptools/#project-urls
+    project_urls={
+        "Source": "https://github.com/cisagov/skeleton-python-library",
+        "Tracker": "https://github.com/cisagov/skeleton-python-library/issues",
```
Comment (jsf9k) [inline setup.py]: These URLs should be updated.

File: setup.py [hunk 3]
```diff
@@ -60,13 +73,14 @@ def package_vars(version_file):
...
         # Specify the Python versions you support here. In particular, ensure
         # that you indicate whether you support Python 2, Python 3 or both.
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.7",
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.6",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
```
Comment (jsf9k) [inline setup.py:82]: What about 3.11?

**Review by jsf9k — COMMENTED** [review]
I can only comment on this PR, but if I could I would request a few minor changes.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mcdonnnj) [reply .github/workflows/build.yml:120]: https://github.com/cisagov/skeleton-python-library/pull/114

Comment (jsf9k) [reply .github/workflows/build.yml:120]: Ah, right.

File: setup.py (hunk 2 above)
Comment (dav3r) [reply setup.py]: @jsf9k this is your PR- you can make the changes here 😆 
```suggestion
        "Source": "https://github.com/cisagov/cyhy-feeds",
        "Tracker": "https://github.com/cisagov/cyhy-feeds/issues",
```

**Review by dav3r — APPROVED** [review]
Aside from what @jsf9k noted, everything else looks fantastic to me.

@jsf9k Reminder that you (or any of us) still need to update the branch protection to remove the outdated checks.

File: setup.py (hunk 2 above)
Comment (jsf9k) [reply setup.py]: My buffoonery knows no bounds.  :laughing:

File: setup.py (hunk 3 above)
Comment (mcdonnnj) [reply setup.py:82]: I'm fine with adding it to the list. When `setup.py` was updated we had not added Python 3.11 support, but the `build.yml` workflow I pulled in does Python 3.11 testing.

Comment (mcdonnnj) [reply setup.py:82]: Please see [6319537](https://github.com/cisagov/cyhy-feeds/pull/43/commits/6319537536141086d853241f6602c3ff764b83bf).

**Review by jsf9k — COMMENTED** [review]
I can only comment on this PR, but I would approve it with gusto if I could!

**Comment by jsf9k** [comment]
:partying_face:
